plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsReader.java

```diff
@@ -176,23 +210,115 @@ else if (columnHandle.getBaseType() == VARBINARY) {
         return new TableStatistics(Estimate.of(recordCount), columnHandleBuilder.buildOrThrow());
     }
 
-    private Map<Integer, Long> readNdvs()
+    private Map<Integer, Long> readNdvs(long snapshotId, Set<Integer> columnIds)
     {
         if (!isExtendedStatisticsEnabled(session)) {
             return ImmutableMap.of();
         }
 
         ImmutableMap.Builder<Integer, Long> ndvByColumnId = ImmutableMap.builder();
-        icebergTable.properties().forEach((key, value) -> {
+        Set<Integer> remainingColumnIds = new HashSet<>(columnIds);
```

Comment:
If you're reading a column that's a projected dereference, will it show up here? Might need to filter those out if they don't have NDV stats, otherwise they'll always stick around in the remaining set.

Reply:
> If you're reading a column that's a projected dereference, will it show up here?

i think this looks only for top level primitive fields -- at least that was the intention.

Reply:
also, note that:
https://github.com/trinodb/trino/blob/0f3ac0795c0273448c09be931cff5116974053dc/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java#L2317